pkg/webui/console/components/events/previews/gateway-uplink-message.js

```diff
@@ -46,22 +46,8 @@ const GatewayUplinkMessagePreview = React.memo(({ event }) => {
       isConfirmed = getByPath(data, 'message.payload.m_hdr.m_type') === 'CONFIRMED_UP'
     }
 
-    if ('rx_metadata' in data.message) {
-      if (data.message.rx_metadata.length > 1) {
-        const gatewayWithHighestSNR = getGatewayWithHighestSNR(data.message.rx_metadata)
-        snr = gatewayWithHighestSNR.snr
-        rssi = gatewayWithHighestSNR.rssi
-      } else {
-        snr = data.message.rx_metadata[0].snr
-        rssi = data.message.rx_metadata[0].rssi
-      }
-    }
-
-    if ('settings' in data.message && 'data_rate' in data.message.settings) {
-      const bandwidth = getByPath(data, 'message.settings.data_rate.lora.bandwidth')
-      const spreadingFactor = getByPath(data, 'message.settings.data_rate.lora.spreading_factor')
-      dataRate = `SF${spreadingFactor}BW${bandwidth / 1000}`
-    }
+    ;({ snr, rssi } = getSignalInformation(data.message))
```

Comment:
```suggestion
    const { snr, rssi } = getSignalInformation(data.message)
```

Reply:
Are you sure this is correct ? The variables are created on L31, and declaring the constants inside the scope of the `if` on L33 does not make them available when they are used on L66/67.

<img width="672" alt="image" src="https://user-images.githubusercontent.com/36161392/186231755-9bd62eaa-0d90-4501-ad55-416ed7329b37.png">

Reply:
You're right indeed. This deconstruction into existing variables is a very uncommon sight but ultimately legitimate. I would write it as:
```js
const signalInfo = getSignalInformation(data.message)
snr = signalInfo.snr
rssi = signalinfo.rssi
```
to avoid the notation which also requires a semicolon.